pkg/repo/migrations/v2_3.go

```diff
@@ -21,7 +21,7 @@ var V2Migration = repo.NewMigration(
 	repo.RepoVersion2,
 	repo.RepoVersion3,
 	func(r repo.FsRepo) error {
-		v, fileCfg, err := readConfig(r)
+		currentCtx, currentCfg, err := readConfig(r)
```

Comment:
we only want to read the configs in the file. Otherwise you will be writing all default configs to the file instead of just the new configs that we want to persist